### ISSUE TITLE
Feature Request: Add third-party authentication options

### ISSUE THREAD

**Issue by Ninjaabhay** [Ninjaabhay/projectSheManages#3]
- **Description**: The current login process is functional, but adding third-party authentication (e.g., Google, GitHub) would make it more convenient for users to sign up and log in.
   - **Solution**: Integrate third-party OAuth providers like Google or GitHub to allow for faster and more secure user authentication.

---

**Comment by yash25hemnani** [comment]
Hey @Ninjaabhay, can you assign me this issue as part of Hacktoberfest?

**Comment by Ninjaabhay** [comment]
hey @yash25hemnani assigned the isuue to you, looking forward to you contribution.

**Comment by yash25hemnani** [comment]
Thank you, I'll get to work. Just wanted to confirm, I will have to raise a PR after 1st October so it gets accepted right?

**Comment by yash25hemnani** [comment]
Also, would you like me to use Passport.js or Clerk for this? Am comfortable with both.

**Comment by Ninjaabhay** [comment]
@yash25hemnani yes if you are looking to count this PR in hactoberfest then it should be raised in october only

**Comment by Ninjaabhay** [comment]
Clerk would be a better choice although you are welcome to use either of them

**Comment by yash25hemnani** [comment]
Okay then, I will go with clerk. Will be raising the PRs in a week.

**Comment by yash25hemnani** [comment]
@Ninjaabhay, do you want me to keep the previous Login and Signup and integrate Clerk on top of it, or can I use Clerk for all of it? And can I make some changes in the UI?

**Comment by Ninjaabhay** [comment]
@yash25hemnani I think integrating the clerk on the previous system wouldn't be a great idea so we should migrate fully on the clerk. And yes can make changes into the ui as well but try to match with the theme of the application.

**Comment by yash25hemnani** [comment]
Alright. Great.

**Comment by yash25hemnani** [comment]
Hey @Ninjaabhay, I noticed there is no Profile Page yet. Could you please raise an issue for it and assign it to me? I am almost done with the Clerk Auth and we need a profile page for complete user data collection.

**Comment by Ninjaabhay** [comment]
Hi @yash25hemnani , Profile page already exists you can see it after logging in.
> Hey @Ninjaabhay, I noticed there is no Profile Page yet. Could you please raise an issue for it and assign it to me? I am almost done with the Clerk Auth and we need a profile page for complete user data collection.

**Comment by yash25hemnani** [comment]
It does appear, but there is no Page or Route for it.

<img width="1529" height="893" alt="Image" src="https://github.com/user-attachments/assets/42da779f-512c-4f71-834b-17eea44e595a" />

**Comment by yash25hemnani** [comment]
Okay, I think I figured it out.

**Comment by yash25hemnani** [comment]
@Ninjaabhay , I have completed the Clerk Auth and all the functionalities are working as well (including the add products, add ideas etc). I have also removed AuthContext and multiple other things as they were unnecessary after implementing Clerk Auth. Clerk has a hook called useUser that provides details of the logged in user directly.

I will be raising a pull request on 1st October and would love to discuss on the code further.

**Comment by Ninjaabhay** [comment]
Hi @yash25hemnani allow me some time traveling right now I'll update you soon.

**Comment by Ninjaabhay** [comment]
> @Ninjaabhay , I have completed the Clerk Auth and all the functionalities are working as well (including the add products, add ideas etc). I have also removed AuthContext and multiple other things as they were unnecessary after implementing Clerk Auth. Clerk has a hook called useUser that provides details of the logged in user directly.
> 
> I will be raising a pull request on 1st October and would love to discuss on the code further.

Cool.
